pkg/tempopb/tempo.proto

```diff
@@ -172,6 +172,8 @@ message PushBytesRequest {
   repeated bytes ids = 3 [(gogoproto.nullable) = false, (gogoproto.customtype) = "PreallocBytes"];
   // search data, length must match traces
   repeated bytes searchData = 4 [(gogoproto.nullable) = false, (gogoproto.customtype) = "PreallocBytes"];
+    // span count per trace
+  repeated bytes spanCounts = 5 [(gogoproto.nullable) = false, (gogoproto.customtype) = "PreallocBytes"];
```

Comment:
if span count is 1 byte then the max spans it can track is 2^8 which is likely not enough in a lot of scenarios. i question whether passing the span count at all is worth it tho.